### ISSUE TITLE
computing sharpness of a minima

### ISSUE THREAD

**Issue by Olwn** [keskarnitish/large-batch-training#3]
hi Nitish,
Can you release the code of computing sharpness? I want to use the metric in my paper.

**Comment by tatsukawa** [comment]
I also want to know how to implement this metric.
Of course, I know the following description at the README.
> The code for computing the sharpness of a minima (Metric 2.1) will be released soon. As is the case with the parametric plots, the code is quite straightforward. The code in Keras' pull-request #3064 along with SciPy's L-BFGS-B optimizer can be used in conjunction to compute the values easily

Reading your paper, I could not understand the (3) equation.
Why do the constraints depend on x? In A is I_n(nxn identity matrix), I may be wrong but the (4) equation roughly is max(f(x±ε(x+Az))). z is the nx1 matrix whose elements are initialized 1.  I think the equation should be max(f(x±εAz)).

In addition, I want to ask you how to decide the range randomly generating A.

**Comment by Billy1900** [comment]
Have you known how to implement the code?
